### DISCUSSION TITLE
[FEATURE] Add support installation on Windows Server 2022

### DISCUSSION THREAD

**Discussion by David-Noble-at-work** [microsoft/winget-cli#1955] [Q&A] [answered]
### Brief description of your issue

Please start by providing a well-advertised and discoverable download and a list of prerequisites with download links. This should be straightforward. I think you will find that most people will say the winget cli is most useful on systems without Microsoft Store availability. It is surprising that a package manager isn't standard on server class machines.

### Steps to reproduce

1. Search the internet for "winget-cli downloads", "winget-cli releases", and "install winget-cli on Windows Server".

2. Find this site and other Microsoft sites that say things like:I

   - Installation on Windows Server isn't supported, 
   - You might be able to install on Windows Server 2022
   - You'll need to instal prerequisites on Windows Server 2022

3. Observe there are third-party articles with instructions and bad links (stuff it seems someone has removed).

4. Observe that this site either hides or obscures anything that would make it possible for someone to discover what's required to install on Windows Server 2022.

5. Suspect that the developers on this site don't care about Windows Server consumers and be sad about using Microsoft Windows at all.

### Expected behavior

Find release notes, download links, and instructions for installing winget-cli on the platform where it is perhaps most useful: Windows Server.

### Actual behavior

- Find release notes, no download links, no instructions, and unanswered questions on support Channels.
- Find instructions from third parties with download links that someone has removed.

### Environment

```shell
Windows Server 2022 machine on Azure that costs quite a bit of money and doesn't include a package manager.
```

**Comment by jedieaston** [comment]
I'm guessing you've already seen #754, but the long story short is that even if installation of winget is supported for Server 2022, it is just about unusable[^1] due to the issues described in #1474. Those are actually upstream OS issues, which may have a workaround specifically for winget at some point but they are mostly waiting to see if it gets resolved in Windows.

There would be very little utility in a package manager where you have to be sitting on the console in order to use it. Once that issue is resolved the other ones should be easy to fix. 

If you're feeling brave, I've been [compiling standalone builds](https://github.com/jedieaston/winget-build) of winget from the master branch that should "just work" on Server 2022. It might be worth me freezing a version that matches the stable release, although it would still be extremely unsupported by Microsoft. They will work just fine over SSH/PSRemoting, and no permissions changes are necessary. 

Edit: tested, my builds do work just fine on Server 2022. They aren't supported by MS, but it's a workaround until this is resolved.

[^1]: Yes, you can force change the permissions on stuff in WindowsApps to make it work. But that's an arcane workaround that will possibly break stuff with appx.

**Reply by cheema-corellian** [reply]
I tested your nightly winget build today with an attempted install of WhatsApp.  It did not work.  See the output below.

==

D:\tmp\AppInstallerCLI>wingetdev.exe install --id 9NKSQGP7F2NH
Found WhatsApp Desktop [9NKSQGP7F2NH] Version Unknown
This package is provided through Microsoft Store. winget may need to acquire the package from Microsoft Store on behalf of the current user.
Version: Unknown
Publisher: WhatsApp Inc.
Description: With WhatsApp on the desktop, you can seamlessly sync all of your chats to your computer so that you can chat on whatever device is most convenient to you.
License: ms-windows-store://pdp/?ProductId=9NKSQGP7F2NH
Privacy Url: https://www.whatsapp.com/legal/#privacy-policy
Agreements:
Category: Social
Pricing: Free
Free Trial: No
Terms of Transaction: https://aka.ms/microsoft-store-terms-of-transaction
Seizure Warning: https://aka.ms/microsoft-store-seizure-warning
Store License Terms: https://aka.ms/microsoft-store-license


The publisher requires that you view the above information and accept the agreements before installing.
Do you agree to the terms?
[Y] Yes  [N] No: Y
An unexpected error occurred while executing the command:
0x803fb104 : unknown error

**Comment by denelon** [comment, marked as answer]
I've converted this to a Discussion so the action of closing the Issue doesn't make the discussion harder to find or contribute to. We've updated the [README.md](https://github.com/microsoft/winget-cli#installing-the-client) with the official status of Windows Package Manager on Windows Server 2022.

**Comment by alexchandel** [comment]
Awesome to create a package manager, whose primary use case is power users and servers, that doesn't run on servers.  Standalone builds are nice, but in practice `0x803fb104 : unknown error` (no MS Store for servers) defeats even the installation of `sysinternals`.

**Reply by jedieaston** [reply]
Sysinternals is blocked because they ship as loose executables (except for the Microsoft Store amalgamation), once there’s portable app support (and zip support) you should be able to install them on machines without the store. 

#140 is the issue to watch.

**Reply by alexchandel** [reply]
What if I want to install them as loose executables?  Why is this inexplicably impossible on servers?  Do you see the problem here?

**Reply by jedieaston** [reply]
You *should* be able to do it now that we have portable app support, I haven't looked at adding sysinternals to the community repo though.

**Comment by cheema-corellian** [comment]
I too would love to see support for winget-cli on Windows Server.

**Reply by denelon** [reply]
We're capturing specific scenarios at:

* https://github.com/microsoft/winget-cli/discussions/2361
